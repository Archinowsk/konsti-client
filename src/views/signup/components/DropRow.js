/* @flow */
import React from 'react'
import { useTranslation } from 'react-i18next'
/* $FlowFixMe: Cannot import `Droppable` because there is no `Droppable` export in `react-beautiful-dnd`. */
import { Droppable, Draggable } from 'react-beautiful-dnd'
import { Link } from 'react-router-dom'
import type { Game } from 'flow/game.flow'

type Props = {
  droppableId: string,
  games: Array<Game>,
  label: string,
}

const DropRow = (props: Props) => {
  const { droppableId, games, label } = props
  const { t } = useTranslation()

  const getListStyle = dragging => {
    if (dragging) return 'dragging'
    else return ''
  }

  return (
    <React.Fragment>
      <p>{label}</p>

      <Droppable droppableId={droppableId}>
        {(provided, snapshot) => (
          <div
            className={`drop-row ${droppableId} ${getListStyle(
              snapshot.isDraggingOver
            )}`}
            ref={provided.innerRef}
          >
            {games.map((game, index) => (
              <Draggable
                key={game.gameId}
                draggableId={game.gameId}
                index={index}
              >
                {(provided, snapshot) => (
                  <div
                    className='draggable-item'
                    ref={provided.innerRef}
                    {...provided.draggableProps}
                    {...provided.dragHandleProps}
                  >
                    {game.title}{' '}
                    <Link to={`/games/${game.gameId}`}>({t('details')})</Link>
                  </div>
                )}
              </Draggable>
            ))}
            {provided.placeholder}
          </div>
        )}
      </Droppable>
    </React.Fragment>
  )
}

export default DropRow
