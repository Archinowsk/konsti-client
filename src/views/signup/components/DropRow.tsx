import React, { Fragment, FunctionComponent, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import { Droppable, Draggable } from 'react-beautiful-dnd';
import { Link } from 'react-router-dom';
import { Game } from 'typings/game.typings';

export interface Props {
  droppableId: string;
  games: readonly Game[];
  label: string;
  showCount: boolean;
}

export const DropRow: FunctionComponent<Props> = (
  props: Props
): ReactElement<typeof Fragment> => {
  const { droppableId, games, label, showCount } = props;
  const { t } = useTranslation();

  const getListStyle = dragging => {
    if (dragging) return 'dragging';
    else return '';
  };

  const getPopularity = (game: Game): string => {
    if (game.popularity >= game.maxAttendance) return 'high-popularity';
    else if (
      game.popularity >= game.maxAttendance / 2 &&
      game.popularity >= game.minAttendance
    )
      return 'medium-popularity';
    else return 'low-popularity';
  };

  return (
    <Fragment>
      <p>
        {label} {showCount && <span>({games.length}/3)</span>}
      </p>

      <Droppable droppableId={droppableId}>
        {(provided, snapshot) => (
          <div
            className={`drop-row ${droppableId} ${getListStyle(
              snapshot.isDraggingOver
            )}`}
            ref={provided.innerRef}
          >
            {games.map((game, index) => (
              <Draggable
                key={game.gameId}
                draggableId={game.gameId}
                index={index}
              >
                {provided => (
                  <Link to={`/games/${game.gameId}`}>
                    <div
                      className={`draggable-item ${getPopularity(game)}`}
                      ref={provided.innerRef}
                      {...provided.draggableProps}
                      {...provided.dragHandleProps}
                    >
                      <p className='signup-game-title break-long'>
                        {game.title}
                      </p>
                      <p className='signup-short-description break-long'>
                        {t(`programType.${game.programType}`)}:{' '}
                        {game.shortDescription
                          ? game.shortDescription
                          : game.gameSystem}
                      </p>
                      <p className='signup-popularity'>
                        {t('playerStatus')}: {t(getPopularity(game))}
                      </p>
                    </div>
                  </Link>
                )}
              </Draggable>
            ))}
            {provided.placeholder}
          </div>
        )}
      </Droppable>
    </Fragment>
  );
};
