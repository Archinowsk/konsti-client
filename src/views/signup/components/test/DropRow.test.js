// @flow
import React from 'react'
import { shallow } from 'enzyme'
import { DropRow } from '../DropRow'
import type { Props } from '../DropRow'

const droppableId = 'testId'
const games = []
const label = 'test label'
const showCount = true

describe('DropRow', () => {
  it('should render correctly', () => {
    const props: Props = {
      droppableId,
      games,
      label,
      showCount,
    }
    const component = shallow(<DropRow {...props} />)
    expect(component).toMatchSnapshot()
  })
})
