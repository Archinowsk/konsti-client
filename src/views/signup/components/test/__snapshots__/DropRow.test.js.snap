// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropRow should render correctly 1`] = `
<Fragment>
  <p>
    test label
     
    <span>
      (
      0
      /3)
    </span>
  </p>
  <Connect(Droppable)
    direction="vertical"
    droppableId="testId"
    ignoreContainerClipping={false}
    isCombineEnabled={false}
    isDropDisabled={false}
    type="DEFAULT"
  >
    <Component />
  </Connect(Droppable)>
</Fragment>
`;
